package/Dockerfile

```diff
@@ -0,0 +1,71 @@
+# If desired this can be set to specific tag by passing an arg
+ARG BCI_VERSION=latest
+FROM registry.suse.com/bci/bci-busybox:${BCI_VERSION} AS final
+
+# Image that provides cross compilation tooling.
+FROM --platform=$BUILDPLATFORM rancher/mirrored-tonistiigi-xx:1.3.0 AS xx
+
+FROM --platform=$BUILDPLATFORM registry.suse.com/bci/golang:1.22 AS build
+
+ARG PROJECT_PATH
+ARG VERSION
+ARG BUILD_DATE
+ARG TAG
+
+COPY --from=xx / /
+
+WORKDIR /app
+COPY . .
+
+ENV CGO_ENABLED=0
+RUN xx-go --wrap && mkdir -p /run/lock
+RUN xx-go mod download && \
+    xx-go build -ldflags "-X=${PROJECT_PATH}/pkg/kuberlr.Version=${VERSION} \
+        -X=${PROJECT_PATH}/pkg/kuberlr.BuildDate=${BUILD_DATE:-$(date +%Y%m%d)} \
+        -X=${PROJECT_PATH}/pkg/kuberlr.Tag=${TAG}" \
+        -o ./bin/kuberlr ./cmd/kuberlr
+RUN xx-verify --static ./bin/kuberlr
+
+FROM registry.suse.com/bci/bci-base:${BCI_VERSION} AS zypper
+
+RUN mkdir /chroot
+COPY --from=final / /chroot/
+
+# The final image does not contain zypper, --installroot is used to
+# install all artefacts within a dir (/chroot) that can then be copied
+# over to a scratch image.
+RUN zypper --non-interactive refresh && \
+    zypper --installroot /chroot -n rm busybox-vi busybox-links && \
+    zypper --installroot /chroot -n in bash-completion && \
+    zypper --installroot /chroot clean -a && \
+    rm -rf /chroot/var/cache/zypp/* /chroot/var/log/zypp/* /chroot/etc/zypp/
+
+COPY --from=build /app/bin/kuberlr /chroot/bin/kuberlr
+RUN cd /chroot/bin && ln -s ./kuberlr ./kubectl
+
+RUN useradd -u 1000 -U -m kuberlr \
+    && cp /etc/passwd /chroot/etc/passwd \
+    && cp /etc/group /chroot/etc/group \
+    && cp -r /home/kuberlr /chroot/home/kuberlr \
+    && echo '. /etc/profile.d/bash_completion.sh' >> /chroot/home/kuberlr/.bashrc \
+    && echo 'alias k="kubectl"' >> /chroot/home/kuberlr/.bashrc \
+    && echo 'alias ks="kubectl -n kube-system"' >> /chroot/home/kuberlr/.bashrc \
+    && echo 'source <(kuberlr completion bash)' >> /chroot/home/kuberlr/.bashrc \
+    && echo 'source <(kubectl completion bash)' >> /chroot/home/kuberlr/.bashrc \
```

Comment:
I think from a UX perspective this was my main concern.

I want to include this by default to make it as usable out of the box. However doing so causes kuberlr to download kubectl right when it starts, instead of JIT when it connects to a server and knows what version to match. So that could be seen as a downside, since a user will see it downloading before having a shell to interact with.

Yet, if we exclude this from the `.bashrc` file then even after it has JIT downloaded kubectl - there's no clean way to make sure the env has bash completion setup for kubectl. Almost like the solution to that would require `kuberlr` be able to source the `kubectl` completion after download.

Reply:
what if we create the bash completion inside of one the build images and then copy the resulting `.bashrc` file into the final image? By doing that we could remove these `source` lines and hence avoid the download of kubectl as soon as you spawn the container